### PR TITLE
HLD - Openconfig support for NAT and respective CLI's

### PR DESCRIPTION
**Pull request by rvasanthm**

### CODE REVIEW

File: doc/mgmt/SONiC_NAT_MgmtSupport_HLD.md [hunk 1]
```diff
@@ -0,0 +1,582 @@
...
+
+### 3.6.2 CLI
+SONiC NAT Click CLI based Configuration and Show Commands will be supported by management framework.
+Refer Section 3.8 from https://github.com/Azure/SONiC/blob/dc5d3a894618bcb07a3c5d2dd488caf3beb7479a/doc/nat/nat_design_spec.md
+
+#### 3.6.2.1 Configuration Commands
+
+###### Add static NAT entry
+
+`static basic <global-ip> <local-ip> {snat | dnat} { twice_nat_id <value> }`
```
Comment (kirankella) [inline doc/mgmt/SONiC_NAT_MgmtSupport_HLD.md]: twice_nat_id is optional parameter in all the commands. So, it has to be within [ ] braces right?

File: doc/mgmt/SONiC_NAT_MgmtSupport_HLD.md [hunk 2]
```diff
@@ -0,0 +1,582 @@
...
+
+```
+sonic# config t
+sonic(config)# nat
+sonic(config-nat) # no static all
+```
+
+###### Create NAT pool
+
+`pool <pool-name> <global-ip-range> <global-port-range>`
```
Comment (kirankella) [inline doc/mgmt/SONiC_NAT_MgmtSupport_HLD.md]: global-port-range is optional parameter.

File: doc/mgmt/SONiC_NAT_MgmtSupport_HLD.md [hunk 3]
```diff
@@ -0,0 +1,582 @@
...
+## 3.4 SyncD
+N/A
+
+## 3.5 SAI
+N/A
+
+## 3.6 User Interface
+### 3.6.1 Data Models
+
+IETF NAT Yang (RFC 8512) is used for the north bound management interface support.
```
Comment (kirankella) [inline doc/mgmt/SONiC_NAT_MgmtSupport_HLD.md:108]: So, now the IETF Yang is not used for the north bound management interface support, right?

File: doc/mgmt/SONiC_NAT_MgmtSupport_HLD.md (hunk 1 above)
Comment (AkhileshSamineni) [inline doc/mgmt/SONiC_NAT_MgmtSupport_HLD.md]: { snat | dnat } is a also a optional parameter.

File: doc/mgmt/SONiC_NAT_MgmtSupport_HLD.md [hunk 4]
```diff
@@ -0,0 +1,582 @@
...
+
+```
+sonic# config t
+sonic(config)# nat
+sonic(config-nat) no bindings
+```
+
+###### Configure NAT zone value on an interface
+
+`nat { zone <zone-value> }`
```
Comment (AkhileshSamineni) [inline doc/mgmt/SONiC_NAT_MgmtSupport_HLD.md]: It has to be like "nat-zone <zone-value>" right.

File: doc/mgmt/SONiC_NAT_MgmtSupport_HLD.md [hunk 5]
```diff
@@ -0,0 +1,582 @@
...
+
+```
+sonic# config t
+sonic(config)# nat
+sonic(config-nat) no pools
+```
+
+###### Create binding between an ACL and a NAT pool
+
+`binding <binding-name> <pool-name> <acl-name> { snat | dnat } { twice_nat_id <value> }`
```
Comment (AkhileshSamineni) [inline doc/mgmt/SONiC_NAT_MgmtSupport_HLD.md]: { snat | dnat } and twice_nat_id are optional parameters.

File: doc/mgmt/SONiC_NAT_MgmtSupport_HLD.md (hunk 1 above)
Comment (ArthiSivanantham) [reply doc/mgmt/SONiC_NAT_MgmtSupport_HLD.md]: Addressed

File: doc/mgmt/SONiC_NAT_MgmtSupport_HLD.md (hunk 2 above)
Comment (ArthiSivanantham) [reply doc/mgmt/SONiC_NAT_MgmtSupport_HLD.md]: Addressed

File: doc/mgmt/SONiC_NAT_MgmtSupport_HLD.md (hunk 1 above)
Comment (ArthiSivanantham) [reply doc/mgmt/SONiC_NAT_MgmtSupport_HLD.md]: Addressed

File: doc/mgmt/SONiC_NAT_MgmtSupport_HLD.md (hunk 4 above)
Comment (ArthiSivanantham) [reply doc/mgmt/SONiC_NAT_MgmtSupport_HLD.md]: Addressed

File: doc/mgmt/SONiC_NAT_MgmtSupport_HLD.md (hunk 5 above)
Comment (ArthiSivanantham) [reply doc/mgmt/SONiC_NAT_MgmtSupport_HLD.md]: Addressed

File: doc/mgmt/SONiC_NAT_MgmtSupport_HLD.md (hunk 3 above)
Comment (rvasanthm) [reply doc/mgmt/SONiC_NAT_MgmtSupport_HLD.md:108]: Yes, not going to use IETF yang. Will be defining openconfig style yang for NAT, will update HLD once yang model is ready.

**Pull request merged**